### ISSUE TITLE
"Cannot GET /<redirect>" on http://<server>/<redirect>

### ISSUE THREAD

**Issue by MailYouLater** [Ylianst/MeshCentral#421]
In order to have an easy url where clients can go to download the meshagent, I've followed the instructions posted at https://github.com/Ylianst/MeshCentral/issues/263#issuecomment-499970552.  Basically this means I created an invitation link that won't expire, and set a redirect that forwards to it. This works great if they go to `https://<server>/<redirect>`, however, people typing the URL in their browser tend to skip the protocol, and end up going to `http://<server>/<redirect>` which simply displays `Cannot GET /<redirect>` instead of redirecting.

I see two ways to fix this issue:
1. Make the `http`->`https` redirect server redirect all traffic to the `https` version of whatever address was typed.
    - This would mean that the `http`->`https` redirect server doesn't need to access the server configuration to look up redirects.
    - This would add an extra redirect, which points to a redirect. If you get too many redirects pointing to redirects pointing to redirects, browsers will complain instead of actually following all the redirects.
2. Make the `http`->`https` redirect server aware of the redirects set in the server configuration, and redirect directly to the intended location.
    - This means one less redirect, so less potential for browsers complaining that a redirect pointed to a redirect, and even if browsers don't complain, it will load a little faster without having to do that extra redirect.
    - This adds complexity to the `http`->`https` redirect server, since it now has to examine the server config to look for redirects.
- There may be other issues with either one of these that I don't know.

**Comment by Ylianst** [comment]
This is a good point. I need to work on http->https redirect for this and probably other links.

**Comment by Ylianst** [comment]
Published MeshCentral v0.3.9-p with a fix for this. I don't redirect all URL's, but I did go thru a list of possible URL that are handled in HTTPS that needed to be redirected from HTTP. Hopefully this is ok with you.

**Comment by MailYouLater** [comment]
I think you misunderstood me.
I have a redirect set in my config for downloading the mesh agent so that customers can install it.
```
    "Redirects": {
        "agentinstaller": "/agentinvite?c=<verylongstring>"
    }
```
If I go to `http://<serverurl>/agentinstaller` I get the 404 error page: "Cannot GET /agentinstaller", but if I go to `https://<serveruser>/agentinstaller` it works fine.

**Comment by Ylianst** [comment]
Oh! Got it. Published MeshCentral v0.3.9-r with that fix. Let me know if it works for you.

**Comment by MailYouLater** [comment]
Works for me now. Thanks.
